pkg/providers/pricing/pricing.go

```diff
@@ -109,16 +110,18 @@ func NewProvider(_ context.Context, pricing pricingiface.PricingAPI, ec2Api ec2i
 
 // InstanceTypes returns the list of all instance types for which either a spot or on-demand price is known.
 func (p *Provider) InstanceTypes() []string {
-	p.mu.RLock()
-	defer p.mu.RUnlock()
+	p.muOnDemand.RLock()
+	p.muSpot.RLock()
+	defer p.muOnDemand.RUnlock()
+	defer p.muSpot.RUnlock()
```

Comment:
I think this is correct since we want to lock both now, as they act like one here